spanner/api/Spanner.Samples.Tests/SpannerFixture.cs

```diff
@@ -164,6 +231,75 @@ private async Task InitializeBackupAsync()
         }
     }
 
+    private async Task InitializeEncryptedBackupAsync()
+    {
+        // Sample backup for encrypted restore test.
+        try
+        {
+            CreateDatabaseWithEncryptionKeyAsyncSample createDatabaseAsyncSample = new CreateDatabaseWithEncryptionKeyAsyncSample();
+            InsertDataAsyncSample insertDataAsyncSample = new InsertDataAsyncSample();
+            await createDatabaseAsyncSample.CreateDatabaseWithEncryptionKeyAsync(ProjectId, InstanceId, FixedEncryptedDatabaseId, KmsKeyName);
+            await insertDataAsyncSample.InsertDataAsync(ProjectId, InstanceId, FixedEncryptedDatabaseId);
+        }
+        catch (Exception e) when (e.ToString().Contains("Database already exists"))
```

Comment:
Could this better be:
```suggestion
        catch (RpcException e) when (e.StatusCode == StatusCode.AlreadyExists)
```

Reply:
I copied this from the other fixed database that is also created and used for this fixture, and I think the choice to check for this specific error message is intentional here, as there are three actions going on in the try block that could throw an `AlreadyExists` error:
1. The `CreateDatabase` command.
2. The creation of the tables in the database.
3. Inserting data into the tables.

Only the failure of the `CreateDatabase` operation should be handled specifically here.

Spanner does however include metadata in the trailers of the errors to indicate what kind of resource already existed or could not be found. We could use that to detect specific types of errors instead of having to rely on checking error messages. We [use this in the Java client](https://github.com/googleapis/java-spanner/blob/70c5b805def9e60775a08abd34d38c8dbd3031e0/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerExceptionFactory.java#L275) to detect `DatabaseNotFound` for example. I'll have a look and see if that is feasible for `AlreadyExists` as well.